storage/netapp_e/na_eseries_amg.py

```diff
@@ -0,0 +1,264 @@
+        required: yes
+    secondaryArrayId:
+        description:
+            - The ID of the secondary array to be used in mirroing process
+        required: yes
+    syncIntervalMinutes:
+        description:
+            - The synchronization interval in minutes
+        required: no
+        default: 10
+    manualSync:
+        description:
+            - Setting this to true will cause other synchronization values to be ignored
+        required: no
+        default: no
+    recoveryWarnThresholdMinutes:
+        description:
+            - Recovery point warning threshold (minutes). The user will be warned when the age of the last good failures point exceeds this value
+        required: no
+        default: 20
+    repoUtilizationWarnThreshold:
+        description:
+            - Recovery point warning threshold 
+        required: no
+        default: 80
+    interfaceType:
+        description:
+            - The intended protocol to use if both Fibre and iSCSI are available.
+        choices:
+            - iscsi
+            - fibre
+        required: no
+        default: null
+    syncWarnThresholdMinutes:
+        description:
+            - The threshold (in minutes) for notifying the user that periodic synchronization has taken too long to complete.
+        required: no
+        default: 10
+    ssid:
+        description:
+            - The ID of the primary storage array for the async mirror action
+        required: yes
+    state:
+        description:
+            - A C(state) of present will either create or update the async mirror group.
+            - A C(state) of absent will remove the async mirror group.
+        required: yes
+"""
+
+EXAMPLES="""
+ - name: AMG create
+   na_eseries_amg:
+     state: present
+     ssid: "{{ ssid }}"
+     api_url: "{{ netapp_api_url }}"
+     api_username: "{{ netapp_api_username }}"
+     api_password: "{{ netapp_api_password }}"
+     description: "{{amg_array_name}}"
+     secondaryArrayId: "{{amg_secondaryArrayId}}"
+"""
+
+
+RETURN="""
+msg:
+    description: Success message
+    returned: success
+    type: string
+    sample: "Desired state met."
+"""
+
+HEADERS = {
+        "Content-Type": "application/json",
+        "Accept": "application/json",
+        }
+
+def request(url, data=None, headers=None, method='GET', use_proxy=True,
+            force=False, last_mod_time=None, timeout=10, validate_certs=True,
+            url_username=None, url_password=None, http_agent=None, force_basic_auth=False, ignore_errors=False):
+    try:
+        r = open_url(url=url, data=data, headers=headers, method=method, use_proxy=use_proxy,
+                     force=force, last_mod_time=last_mod_time, timeout=timeout, validate_certs=validate_certs,
+                     url_username=url_username, url_password=url_password, http_agent=http_agent,
+                     force_basic_auth=force_basic_auth)
+    except urllib2.HTTPError, err:
```

Comment:
For python3 compatibility, exception handling syntax needs to look a little different.  Which syntax to use depends on whether we also have to support python2.4.  If we do, then the syntax will be:
``` python

from ansible.module_utils.pycompat24 import get_exception
# [..] Do stuff
try:
    # [..] Do more stuff
except urllib2.HTTPError:
    err = get_exception()
    r = err.fp
```

If we don't have to support python-2.4, the syntax is simply the python3 syntax:
``` python
try:
    # [..] Do stuff
except urllib2.HTTPError as err:
    r = err.fp
```